packages/grafana-data/src/types/logs.ts

```diff
@@ -225,36 +225,44 @@ export interface DataSourceWithSupplementaryQueriesSupport<TQuery extends DataQu
   /**
    * Returns an observable that will be used to fetch supplementary data based on the provided
    * supplementary query type and original request.
+   * @deprecated Use getSupplementaryQueryRequest() instead
    */
-  getDataProvider(
+  getDataProvider?(
     type: SupplementaryQueryType,
     request: DataQueryRequest<TQuery>
   ): Observable<DataQueryResponse> | undefined;
+  /**
+   * Given a DataQueryRequest returns a new DataQueryRequest to be used to fetch supplementary data.
```

Comment:
I don't understand this first part of the comment.

Reply:
I mean with this that you receive a `DataQueryRequest` and return another `DataQueryRequest` to fetch the supplementary data. Is there a specific part of the comment that you find particularly unclear?

If it's the "given" part, it's like saying "given an array of numbers, returns odd numbers"